### PR TITLE
Fixes #7894. Sprinting units can no longer autohit adjacent buildings

### LINKED ISSUE #7894: [Issue] Meks using sprinting movement can still make weapon and physical attacks against adjacent buildings, which hit with 100% success rate
### Brief Description *

After sprinting a mek adjacent to a building, that mek can declare weapon and physical attacks against the building, showing a 100% chance of success, and actually hitting with 100% chance.

Image

TO:AR 6th printing page 16 rules for Sprinting say a mek that sprints may make weapon attacks, but they must do no damage, and may not make physical attacks.

Image

Weapon and physical attacks against adjacent buildings after sprinting succeed and do damage

Image

### 3. Steps to Reproduce

In MegaMek enable sprinting and start a game on a map with buildings:
Sprint a mek adjacent to a building.
In the weapon phase attack the adjacent building.
Observe hit in round report.
In the physical phase attack the adjacent building.
Observe the hit in the round report.

The attached save has a Griffin already adjacent to a building.

### Attach Files

[attack_adjacent_buildings_issue.sav.gz](https://github.com/user-attachments/files/24719952/attack_adjacent_buildings_issue.sav.gz)

### Severity *

Low (Minor/Nuisance): Minor glitches or cosmetic issues that don't affect gameplay and occur rarely.

### MegaMek Suite Version *

50.11

### Operating System *

Windows 10

### Java Version *

    Java Vendor: Eclipse Adoptium     Java Version: 17.0.13

### Final Verification

- [x] I confirm this is a single, unique issue that hasn't been reported before
- [x] I have included all necessary information and files to help reproduce this issue
- [x] I have asked on MegaMek Discord about this issue
- [x] I have confirmed this issue is being opened on the correct repository: MegaMek, MegaMekLab, or MekHQ

### PR DESCRIPTION
**Pull request by TenkawaBC**
Fixes #7894 

Add logic to the toHitIsAutomatic to not return an automatic hit for adjacent buildings if they sprinted.

To do this, when checking for an adjacent building, it also checks if the unit sprinted, and fails if it is true. This results in a null return, which is the correct behavior to indicate it is not an automatic hit.

### CODE REVIEW

File: megamek/src/megamek/common/actions/compute/ComputeToHit.java [hunk 1]
```diff
@@ -1007,10 +1007,10 @@ private static String toHitIsAutomatic(Game game, Entity weaponEntity, Targetabl
               (targetType == Targetable.TYPE_FUEL_TANK_IGNITE) ||
               (target.isBuildingEntityOrGunEmplacement());
 
-        if ((distance == 1) && isBuilding) {
+        if ((distance == 1) && isBuilding && !(weaponEntity.moved == EntityMovementType.MOVE_SPRINT)) {
```
[the remark this answers is not in the document]
Comment (TenkawaBC) [reply megamek/src/megamek/common/actions/compute/ComputeToHit.java]: Addressed in latest commit.

[the remark this answers is not in the document]
Comment (TenkawaBC) [reply megamek/src/megamek/common/actions/compute/ComputeToHit.java]: Changed in latest commit, as I am now using 2 different conditions. For readability I found it easier to do it the other way, but it is the effectively same thing.

[the remark this answers is not in the document]
Comment (TenkawaBC) [reply megamek/src/megamek/common/actions/compute/ComputeToHit.java]: Edge case, and I'm not versed enough in test case creation to be able to create the test for this.

File: megamek/src/megamek/common/actions/compute/ComputeToHit.java [hunk 2]
```diff
@@ -1007,10 +1007,10 @@ private static String toHitIsAutomatic(Game game, Entity weaponEntity, Targetabl
               (targetType == Targetable.TYPE_FUEL_TANK_IGNITE) ||
               (target.isBuildingEntityOrGunEmplacement());
 
-        if ((distance == 1) && isBuilding) {
+        if ((distance == 1) && isBuilding && (weaponEntity.moved != EntityMovementType.MOVE_SPRINT && weaponEntity.moved != EntityMovementType.MOVE_VTOL_SPRINT)) {
```
Comment (psikomonkie) [inline megamek/src/megamek/common/actions/compute/ComputeToHit.java]: Should a HHW carried by a Mek that sprinted be able to automatically hit an adjacent building? Consider using `ae` instead of `weaponEntity` so the movement check is on the entity making the attack instead of the unit that physically contains the weapon.

EDIT: Also congratulations, you're the first person to edit the file this year. Can you update the copyright year in the header to have `-2026`? Thanks!

Comment (TenkawaBC) [reply megamek/src/megamek/common/actions/compute/ComputeToHit.java]: Addressed in 8c9f4da

**Comment by GiovanniBertuccio** [comment]
I'm not sure if exact conformity of the rules can happen here, but technically a mek that sprints can still attack with weapons - they just can't hit. That's a separate issue, but I wonder if this fix would later have to be changed to implement that behavior.

**Comment by TenkawaBC** [comment]
> I'm not sure if exact conformity of the rules can happen here, but technically a mek that sprints can still attack with weapons - they just can't hit. That's a separate issue, but I wonder if this fix would later have to be changed to implement that behavior.

This does not interfere with any logic around that. It just prevents an auto-hit overriding a to-hit of impossible when adjacent to a building and sprinting.

**Review by psikomonkie — APPROVED** [review]
Nice!
